### PR TITLE
1474 counters for workspaces and companies

### PR DESCRIPTION
**Pull request by romanesko**

### CODE REVIEW

File: twake/backend/node/src/core/platform/services/counter/provider.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      if (actual != currentValue) {
+        await this.increase(pk, actual - currentValue);
+        currentValue = actual;
+      }
+      lastRevised.calls = 0;
+    } else {
+      lastRevised.calls++;
+    }
+
+    lastRevised.period = now;
```
Comment (RomaricMourgues) [inline twake/backend/node/src/core/platform/services/counter/provider.ts:64]: Shouldn't this be after line 59 ?

File: twake/backend/node/src/services/workspaces/entities/workspace_counters.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  primaryKey: [["id"], "counter_type"],
+  type: TYPE,
+})
+export class WorkspaceCounterEntity extends CounterEntity {
+  @Column("id", "timeuuid")
+  id: string;
+
+  @Column("counter_type", "string")
+  counter_type: string;
+}
```
Comment (RomaricMourgues) [inline twake/backend/node/src/services/workspaces/entities/workspace_counters.ts:16]: I don't understand why we don't have the "value" field in the counter entities ?

Comment (RomaricMourgues) [reply twake/backend/node/src/services/workspaces/entities/workspace_counters.ts:16]: Nevermind I just saw the `extends CounterEntity`, this is great 👍

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]

File: twake/backend/node/src/core/platform/services/counter/provider.ts (hunk 1 above)
Comment (RomaricMourgues) [reply twake/backend/node/src/core/platform/services/counter/provider.ts:64]: @romanesko this is the only change I need :)

Comment (romanesko) [reply twake/backend/node/src/core/platform/services/counter/provider.ts:64]: 🤔…. 😑… 🤔 … 😳 … 🙈 … 🤦‍♂️ …. 😬

Comment (RomaricMourgues) [reply twake/backend/node/src/core/platform/services/counter/provider.ts]: Best emojiset ever !

**Pull request merged**